### ISSUE TITLE
Inconsistent vertical navigation menu active states

### ISSUE THREAD

**Issue by richtabor** [WordPress/gutenberg#46681]
## What problem does this address?
We have a few instances of a vertical navigation menu throughout the editor, though we're using different styles for each to indicate an actively selected state. 

We've historically used the `$gray-900` background for indicating an active state, though the newest implementation (Patterns/Media Inserter) uses the blue on lighter blue. 

## What is your proposed solution?
Let's align on one selected state for these components to tighten up the experience. 

## Visual 
<img width="1518" alt="CleanShot 2022-12-20 at 11 33 02@2x" src="https://user-images.githubusercontent.com/1813435/208717647-f186561d-913b-4809-83ab-e1977ebaa39c.png">

**Comment by jasmussen** [comment]
Yep, good issue, I don't think one existed. #44329 was one previous attempt at unifying some things. The blue underline on the left tab is actually a regression from #46276, my bad, I'll take a look.

I would differentiate with the patterns flyout, though, as that isn't necessarily entirely the same component. To me that component has more in common with the styles drilldown:
<img width="310" alt="Screenshot 2022-12-27 at 11 12 16" src="https://user-images.githubusercontent.com/1204802/209651447-fbf4e9ec-788a-489c-a5ef-43bf7bbed78c.png">

Of course that doesn't mean the two can't share DNA, just that one is more compressed at least at the moment.
